modules/standard/online/log_controller.py

```diff
@@ -1,56 +1,75 @@
+        return "Your logoff message has been cleared"
+
+    @command(command="logon", params=[Any("logon_message", is_optional=True)], access_level="member",
+             description="Check your current logon message or set a new one.")
+    def set_or_check_logon(self, request: CommandRequest, logon_message):
+        if logon_message is None:
+            current_logon = self.get_logon(request.sender.char_id)
+            if current_logon:
+                return "%s's login message is: %s" % (request.sender.name, current_logon)
+            else:
+                return "Your logon message has not been set"
         else:
-            self.db.query_single("INSERT INTO log_messages (char_id, logoff) VALUES(?, ?);",
-                                 [request.sender.char_id, logoff_message])
-        return "Your new logoff message is: %s" % logoff_message
+            if self.db.query_single("SELECT logon FROM log_messages WHERE char_id=?;", [request.sender.char_id]):
+                self.db.query_single("UPDATE log_messages SET logon=? WHERE char_id=?;",
+                                     [logon_message, request.sender.char_id])
+            else:
+                self.db.query_single("INSERT INTO log_messages (char_id, logon) VALUES(?, ?);",
+                                     [request.sender.char_id, logon_message])
+            return "Your new logon message is: %s" % logon_message
 
-    def get_logon(self, char_id):
-        content = self.db.query_single("SELECT logon FROM log_messages WHERE char_id=?", [char_id])
-        if content:
-            return "<grey>" + content.get("logon") + "<end>"
+    @command(command="logoff", params=[Any("logoff_message", is_optional=True)], access_level="member",
+             description="Check your current logoff message or set a new one")
+    def set_or_check_logoff(self, request: CommandRequest, logoff_message):
+        if logoff_message is None:
+            current_logoff = self.get_logoff(request.sender.char_id)
+            if current_logoff:
+                return "%s's logoff message is: %s" % (request.sender.name, current_logoff)
+            else:
+                return "Your logoff message has not been set"
         else:
-            return ""
+            if self.db.query_single("SELECT logoff FROM log_messages WHERE char_id=?;", [request.sender.char_id]):
+                self.db.query_single("UPDATE log_messages SET logoff=? WHERE char_id=?;",
+                                     [logoff_message, request.sender.char_id])
+            else:
+                self.db.query_single("INSERT INTO log_messages (char_id, logoff) VALUES(?, ?);",
+                                     [request.sender.char_id, logoff_message])
+            return "Your new logoff message is: %s" % logoff_message
+
+    def get_logon(self, char_id):
+        row = self.db.query_single("SELECT * FROM log_messages WHERE char_id=?", [char_id])
+        if row is not None:
+            if row.logon:
```

Comment:
you could combine these into a single check:
`if row and row.logon:`

not a requirement, and I'm fine with them either way